integration-docs: Update for new format (Harbor, Greenhouse, JSON, Jotform)

PieterCK opened the pull request:
As the changes were mostly to the docs format and not content, these are grouped together in a PR that updates 4 doc pages.
As part of https://github.com/zulip/zulip/issues/29592, update the following integration documentation pages:

**Screenshots and screen captures:**

<details>

<summary>

[Harbor](https://zulip.com/integrations/doc/harbor)
</summary>

![image](https://github.com/zulip/zulip/assets/108744694/9df736fd-48c2-49e2-a933-8a5cd50f0472)
</details>

<details>

<summary>

[Greenhouse](https://zulip.com/integrations/doc/greenhouse)
</summary>

![image](https://github.com/zulip/zulip/assets/108744694/1d49cd68-68dc-4cd1-b753-84dcd232f36a)
</details>

<details>
<summary>

[JSON](https://zulip.com/integrations/doc/json)
</summary>

![image](https://github.com/zulip/zulip/assets/108744694/5a1514eb-f928-4939-bafb-c169f6657c51)
</details>

<details>
<summary>

[Jotform](https://zulip.com/integrations/doc/jotform)
</summary>

![image](https://github.com/zulip/zulip/assets/108744694/1af26528-c108-4c4a-bcd1-dcb207043fe4)
</details>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Is this one ready for @adnan-td and @Niloth-p to take a look?

adnan-td commented on zerver/webhooks/jotform/doc.md:
```diff
@@ -14,6 +18,18 @@ Get Zulip notifications for your Jotform responses!
 1. Add the above constructed **url** to the field, and click the
    **COMPLETE INTEGRATION** button.
 
```
In Harbor, Greenhouse and Jotform docs we can use "generated" instead of "constructed" for the URL.

adnan-td commented on zerver/webhooks/jotform/doc.md:
```diff
@@ -1,10 +1,14 @@
 
-1. {!generate-integration-url.md!}
+1. {!generate-webhook-url-basic.md!}
    By default, the integration will use the form's title as the topic.
```
This line feels out of place

![image](https://github.com/zulip/zulip/assets/78212328/d3082b83-c19a-4c63-a565-bc77a4ec9570)

adnan-td reviewed (commented):
Left comments

PieterCK commented:
Update: Edited phrasing, addressed review from @adnan-td , and added commit description for some commits.

@Niloth-p Could you please take a look at this one as well? I've also applied the comment you left at #30725 to here. Let me know if there are any oversights or any other suggestions. Thanks!

Niloth-p commented on zerver/webhooks/jotform/doc.md:
```diff
@@ -1,19 +1,38 @@
+1. Add the **url** generated above to the field, and click the
    **COMPLETE INTEGRATION** button.
 
+!!! warn ""
```
I think this can be placed near the original location instead, since the information is relevant to the URL generation.
Just indent it to make it appear as part of the step.
Also, we could use a **tip** instead of a **warning**?

Niloth-p reviewed (commented):
Lgtm.

3 documents for integrations that do not support event filtering are using the {!event-filtering-additional-feature.md!} macro, they can be removed, as per Lauryn's reminder on CZO.

PieterCK commented:
Update: addressed review

@Niloth-p & @adnan-td  - Thanks for reviewing!

Niloth-p reviewed (commented):
Just 1 more tiny change that I noticed.

PieterCK commented:
@Niloth-p Updated!

alya commented:
@laurynmm this one should be ready for you to take a look.

laurynmm commented:
@PieterCK - These updates look good! I pushed a quick update to the Jotform doc that cleans up the instructions a bit more. We also should note we need to update the icon/logo for that integration. Going to go ahead and mark for integration review. Thanks for working on these updates!

<details>
<summary>Updated Jotform doc</summary>

[Current documentation](https://zulip.com/integrations/doc/jotform)
![Screenshot from 2024-07-12 20-54-19](https://github.com/user-attachments/assets/cee36d04-dcf6-4cfa-97bc-2d8b76312654)
</details>

timabbott commented:
Merged, thanks @PieterCK and everyone else who helped review this! I love the teamwork on this sweep.
